src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/TaskListNotesModal.tsx

```diff
@@ -0,0 +1,271 @@
+                        getOptionLabel={(option) => option.label}
+                        onChange={(_, value) => onChange(value.map((v) => v.id))}
+                        value={value.map((v) => options.find((o) => o.id === v)!)}
+                        renderInput={(params) => <TextField {...params} variant="standard" placeholder="Select A User" />}
+                      />
+                    )}
+                  />
+                </FormControl>
+              </Grid>
+              <Grid item xs={12} md={4}>
+                <FormControl fullWidth>
+                  <FormLabel>Deadline (YYYY-MM-DD)</FormLabel>
+                  <Controller
+                    name="deadline"
+                    control={control}
+                    rules={{ required: true }}
+                    render={({ field: { onChange, value } }) => (
+                      <DatePicker
+                        inputFormat="yyyy-MM-dd"
+                        onChange={onChange}
+                        className={'padding: 10'}
+                        value={value}
+                        renderInput={(params) => <TextField autoComplete="off" {...params} />}
+                      />
+                    )}
+                  />
+                </FormControl>
+              </Grid>
+              <Grid item xs={12} md={12}>
+                <FormControl fullWidth>
+                  <FormLabel>Notes</FormLabel>
+                  <Controller
+                    name={'notes'}
+                    control={control}
+                    rules={{ required: true }}
+                    render={({ field: { onChange, value } }) => (
+                      <TextField
+                        required
+                        onChange={onChange}
+                        value={value}
+                        multiline
+                        rows={5}
+                        inputProps={{
+                          maxlength: isUnderWordCount(value, 250) ? null : 0
+                        }}
+                        helperText={`${countWords(value)}/250 words`}
+                        error={!isUnderWordCount(value, 250)}
+                      />
+                    )}
+                  />
+                </FormControl>
+              </Grid>
+            </Grid>
+            <Box textAlign="right" gap={2} sx={{ mt: 2 }}>
+              <NERFailButton
+                variant="outlined"
+                onClick={() => {
+                  setIsEditMode(false);
+                }}
+                sx={{ mx: 1 }}
+              >
+                Cancel
+              </NERFailButton>
+              <NERSuccessButton variant="contained" disabled={auth.user!.role === 'GUEST'} type="submit" sx={{ mx: 1 }}>
+                Confirm
+              </NERSuccessButton>
+            </Box>
+          </form>
+        </DialogContent>
+      </Dialog>
+    );
+  };
+
+  return isEditMode ? renderEditModal() : renderModal();
```

Comment:
I think you can do the thing where you make the names EditModal and ViewModal, and then call them here like react components e.g. <EditModal/>